pkg/reconciler/awscloudwatchlogsource/adapter.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright (c) 2020 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awscloudwatchlogsource
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	"knative.dev/eventing/pkg/reconciler/source"
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/kmeta"
+
+	"github.com/triggermesh/aws-event-sources/pkg/apis/sources/v1alpha1"
+	"github.com/triggermesh/aws-event-sources/pkg/reconciler/common"
+	"github.com/triggermesh/aws-event-sources/pkg/reconciler/common/resource"
+)
+
+const (
+	envPollingInterval = "POLLING_INTERVAL"
+)
+
+// adapterConfig contains properties used to configure the source's adapter.
+// These are automatically populated by envconfig.
+type adapterConfig struct {
+	// Container image
+	Image string `default:"gcr.io/triggermesh/awscloudwatchsource"`
+	// Configuration accessor for logging/metrics/tracing
+	configs source.ConfigAccessor
+}
+
+// adapterDeploymentBuilder returns an AdapterDeploymentBuilderFunc for the
+// given source object and adapter config.
+func adapterDeploymentBuilder(src *v1alpha1.AWSCloudWatchLogSource, cfg *adapterConfig) common.AdapterDeploymentBuilderFunc {
+	adapterName := common.AdapterName(src)
+
+	return func(sinkURI *apis.URL) *appsv1.Deployment {
+		name := kmeta.ChildName(adapterName+"-", src.Name)
+
+		var sinkURIStr string
+		if sinkURI != nil {
+			sinkURIStr = sinkURI.String()
+		}
+
+		var pollingInterval string
+		if src.Spec.PollingFrequency != nil {
+			pollingInterval = *src.Spec.PollingFrequency
+		} else {
+			// Defaulting to 5m which should keep things within the free tier
+			pollingInterval = "5m"
+		}
```

Comment:
This can be simplified:
```suggestion
		pollingInterval := "5m"
		if src.Spec.PollingFrequency != nil {
			pollingInterval = *src.Spec.PollingFrequency
		}
```

Reply:
Done in #239